src/ndlbulkdownload/args.py

```diff
@@ -50,6 +50,13 @@ def arg_parser():
                         action='append',
                         help='Add query param key/value pair')
 
+    parser.add_argument('--output_dir',
```

Comment:
Looking over this again, I think want to keep the long-format parameters to use hyphen separators, and allow `argparse` to set the destination in the expected destination variable using a more 'pythonic' variable name which uses an underscore.

Additionally, would prefer to use the full words rather than an abbreviation.

--output-directory

I think it would also be worth while to use a short-argument as well. -O perhaps.

```suggestion
    parser.add_argument('--output_dir',
     parser.add_argument('-O', '--output-directory',
```

This suggested change will also require refactoring within the check_and_set convenience method.

I'm going to start collecting these changes in a private branch as I also want to work through some of my broader ideas for improving the project's lint setup.